src/platforms/mesa/server/x11/X11_resources.cpp

```diff
@@ -55,3 +58,22 @@ std::shared_ptr<::Display> mx::X11Resources::get_conn()
     connection = new_conn;
     return new_conn;
 }
+
+void mx::X11Resources::set_output_config_for_win(Window win, mg::DisplayConfigurationOutput* configuration)
+{
+    output_configs[win] = configuration;
+}
+
+void mx::X11Resources::clear_output_config_for_win(Window win)
+{
+    output_configs.erase(win);
+}
+
+std::experimental::optional<mg::DisplayConfigurationOutput*> mx::X11Resources::get_output_config_for_win(Window win)
```

Comment:
If you're going to return a `std::experimental::optional`, I'd mildly prefer `std::experimental::optional<mg::DisplayConfigurationOutput&>` to the pointer version.

Reply:
I don't think that's allowed. When I try, I get
```
/home/wmww/code/mir/src/platforms/mesa/server/x11/X11_resources.cpp:72:122:   required from here
/usr/include/c++/8/experimental/optional:461:15: error: non-static data member ‘std::experimental::fundamentals_v1::_Optional_base<const mir::graphics::DisplayConfigurationOutput&, false>::<unnamed union>::_M_payload’ in a union may not have reference type ‘std::experimental::fundamentals_v1::_Optional_base<const mir::graphics::DisplayConfigurationOutput&, false>::_Stored_type’ {aka ‘const mir::graphics::DisplayConfigurationOutput&’}
```
I also found [this](https://foonathan.net/blog/2018/07/12/optional-reference.html) on the topic.